gossipd/routing.c

```diff
@@ -1208,6 +1210,7 @@ bool routing_add_channel_update(struct routing_state *rstate,
 	struct unupdated_channel *uc;
 	u8 direction;
 	struct amount_sat sat;
+	bool spam = false;
```

Comment:
As a rule, I prefer doing this assignment inside an else.

That way, if there's a jump over the code which sets it (or one is introduced in future!), the compiler will warn about it (at least, with optimization set), rather than it silently being left at false.

It's just a preference though: sometimes there's no other obvious place to set it.

Reply:
I did it this way first, but was conditionally assigning in several different places. @vincenzopalazzo thought this was cleaner for the reader to understand.  I'm okay either way, but I think he has point.

Reply:
Yeah this was my suggestion just because there was several else branch introduces just for the `spam = false` assignment, like

```c
if (..) {
  if (...) {
   spam = true
  } else 
   spam = false
} else 
   spam = false
```

this can introduce confusion if we introduce other branches in the future. However, I always think that @rustyrussell is right with these code-style suggestions, and I'm very curious to know why this "not assigned at declaration time" is a preference.

We can learn something new :)

Reply:
Reverting to the uninitialized method.